examples/java/src/main/java/org/apache/beam/examples/timeseries/TimeSeriesExampleToBigTable.java

```diff
@@ -0,0 +1,182 @@
+
+    PCollection<KV<TimeSeriesData.TSKey, TimeSeriesData.TSAccum>> downSampled =
+        readData.apply(new ExtractAggregates(configuration)).apply(new GetWindowData());
+
+    PCollection<KV<TimeSeriesData.TSKey, TimeSeriesData.TSAccum>> weHaveOrder =
+        downSampled.apply(new OrderOutput(configuration));
+
+    // ------------ OutPut Data as Logs and TFRecords--------
+
+    // This transform is purely to allow logged debug output, it will fail with OOM if large dataset is used.
+    weHaveOrder.apply(new DebugSortedResult());
+    // Write to Bigtable
+
+    // tf.Example output
+    weHaveOrder
+        .apply(ParDo.of(new GetValueFromKV<>()))
+        .apply(new TSAccums.OutPutToBigTable())
+        .apply(CloudBigtableIO.writeToTable(config));
+
+    // Create 3 different window lengths for the TFSequenceExample
+    weHaveOrder
+        .apply(
+            new TSAccumToFixedWindowSeq(
+                "Sequence of 1 Min", configuration, Duration.standardMinutes(1)))
+        .apply(ParDo.of(new GetValueFromKV<>()))
+        .apply(new TSAccumSequences.OutPutToBigTable())
+        .apply(CloudBigtableIO.writeToTable(config));
+
+    weHaveOrder
+        .apply(
+            new TSAccumToFixedWindowSeq(
+                "Sequence of 5 Min", configuration, Duration.standardMinutes(5)))
+        .apply(ParDo.of(new GetValueFromKV<>()))
+        .apply(new TSAccumSequences.OutPutToBigTable())
+        .apply(CloudBigtableIO.writeToTable(config));
+
+    weHaveOrder
+        .apply(
+            new TSAccumToFixedWindowSeq(
+                "Sequence of 15 Min", configuration, Duration.standardMinutes(15)))
+        .apply(ParDo.of(new GetValueFromKV<>()))
+        .apply(new TSAccumSequences.OutPutToBigTable())
+        .apply(CloudBigtableIO.writeToTable(config));
+
+    p.run();
+  }
+
+  /** Simple data generator that creates some dummy test data for the timeseries examples. */
+  public static class SinWaveSample {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SinWaveSample.class);
+
+    public static List<TimeSeriesData.TSMultiVariateDataPoint> generateSinWave() {
+
+      double y;
+      double yBase = 1;
+      double scale = 20;
+
+      List<TimeSeriesData.TSMultiVariateDataPoint> dataPoints = new ArrayList<>();
+
+      for (int k = 0; k < 10; k++) {
+
+        Instant now = Instant.parse("2018-01-01T00:00Z");
+
+        for (int i = 0; i < 1000; i++) {
+
+          if (!((i % 10 == 0))) {
+
+            Instant dataPointTimeStamp = now.plus(Duration.standardSeconds(i));
+
+            y = (yBase - Math.sin(Math.toRadians(i)) * scale);
+
+            TimeSeriesData.TSMultiVariateDataPoint mvts =
+                TimeSeriesData.TSMultiVariateDataPoint.newBuilder()
+                    .setKey(TimeSeriesData.TSKey.newBuilder().setMajorKey("Sin-" + k).build())
+                    .putData("x", TimeSeriesData.Data.newBuilder().setIntVal(i).build())
```

Comment:
Is it possible to add factory methods to `TimeSeries.Data`?  You call `.newBuilder()` a number of times in the code, i think something like `TimeSeriesData.newInt()` that calls `newBuilder().setIntVal().build()` would make it more readable

Reply:
I do have utility class for each of the main protos. In similar style to Timestamp with protobuff which has utils in Timestamps. Each of the protos have a (s) util class. 

I do have a factory for creating a TimeSeriesData.Data object, which I will reuse here. 

https://github.com/rezarokni/beam/blob/c492cbd74981e2c9c3ae206cb7a7b83b3a487e11/examples/java/src/main/java/org/apache/beam/examples/timeseries/utils/TSDatas.java#L241